R/tcplSubsetChid.R

```diff
@@ -3,154 +3,169 @@
+#' made, the samples corresponding to to chemical-wise hit call are logically
+#' ordered using the fit category, the number of the flags, and the modl_ga,
 #' then the first sample for every chemical is selected.
-#' 
-#' The \code{flag} param can be used to specify a subset of flags to be used in 
+#'
+#' The \code{flag} param can be used to specify a subset of flags to be used in
 #' the flag count. Leaving \code{flag} TRUE utilize all the available flags.
-#' Setting \code{flag} to \code{FALSE} will do the subsetting without 
+#' Setting \code{flag} to \code{FALSE} will do the subsetting without
 #' considering any flags.
-#' 
+#'
 #' @examples
-#' ## Store the current config settings, so they can be reloaded at the end 
+#' ## Store the current config settings, so they can be reloaded at the end
 #' ## of the examples
 #' conf_store <- tcplConfList()
 #' tcplConfExample()
-#' 
+#'
 #' ## Load the example level 5 data
 #' d1 <- tcplLoadData(lvl = 5, fld = "aeid", val = 797)
 #' d1 <- tcplPrepOtpt(d1)
-#' 
+#'
 #' ## Subset to an example of a duplicated chid
 #' d2 <- d1[chid == 20182]
-#' d2[ , list(m4id, hitc, fitc, modl_ga)]
-#' 
-#' ## Here the consensus hit-call is 1 (active), and the fit categories are 
+#' d2[, list(m4id, hitc, fitc, modl_ga)]
+#'
+#' ## Here the consensus hit-call is 1 (active), and the fit categories are
 #' ## all equal. Therefore, if the flags are ignored, the selected sample will
 #' ## be the sample with the lowest modl_ga.
-#' tcplSubsetChid(dat = d2, flag = FALSE)[ , list(m4id, modl_ga)]
-#' 
+#' tcplSubsetChid(dat = d2, flag = FALSE)[, list(m4id, modl_ga)]
+#'
 #' ## Reset configuration
 #' options(conf_store)
-#' 
-#' @return A data.table with a single sample for every given chemical-assay 
-#' pair. 
-#' 
+#'
+#' @return A data.table with a single sample for every given chemical-assay
+#' pair.
+#'
 #' @seealso \code{\link{tcplPrepOtpt}}
-#' 
+#'
 #' @import data.table
 #' @export
 
-tcplSubsetChid <- function(dat, flag = TRUE, type = "mc") {
-  
+tcplSubsetChid <- function(dat, flag = TRUE, type = "mc", export_ready = TRUE) {
   ## Variable-binding to pass R CMD Check
   chit <- hitc <- aeid <- casn <- fitc <- fitc.ordr <- m4id <- nflg <- NULL
   chid <- logc <- minc <- NULL
-  
-  if (!type %in% c("mc","sc")) {
+
+  if (!type %in% c("mc", "sc")) {
     stop("type must be sc (single concentration) or mc (multi-concentration)")
   }
-  
-  if(type == "mc"){
+
+  # if only using export ready, filter data by export ready status
+  if (export_ready) {
+    # try catch here to make sure db has available annotation information
+    tryCatch(
+      {
+        assay_info <- tcplLoadAeid(fld = "aeid", val = dat$aeid, add.fld = "export_ready")
+      },
+      error = function(error) {
+        message(error)
+        if (grepl("Not all given fields available in query", error)) stop(paste0("\n'export_ready' column is not available in ", options()$TCPL_DB, ".  Consider setting export_ready to FALSE."))
```

Comment:
I think brackets are missing from the if statement on line 82.

Reply:
Curly brackets are only needed when there is more than one statement per inside the if otherwise they can follow the format `if (condition) statement else statement`

Reply:
Thanks for clearing that up!